src/Lexer.php

```diff
@@ -396,6 +402,64 @@ private function lexExpression(): void
         }
     }
 
+    private function stripcslashes(string $str, string $quoteType): string {
+        $result = '';
+        $length = strlen($str);
+        $specialChars = [
+            'f' => "\f",
+            'n' => "\n",
+            'r' => "\r",
+            't' => "\t",
+            'v' => "\v",
+        ];
+
+        for ($i = 0; $i < $length; $i++) {
+            if ($str[$i] === '\\' && $i + 1 < $length) {
+                $nextChar = $str[$i + 1];
+                if (isset($specialChars[$nextChar])) {
+                    $result .= $specialChars[$nextChar];
+                    $i++;
+                } elseif ($nextChar === '#') {
+                    $result .= "#";
+                    $i++;
+                } elseif ($nextChar === '\\') {
+                    if ($quoteType === "'" && (!($i + 2 < $length && isset($specialChars[$str[$i + 2]])))) {
+                        $this->deprecations[] = sprintf("Character %s at position %d does not require to be escaped.", $nextChar, $i + 2);
+                    }
+                    $result .= '\\';
+                    $i++;
+                } elseif ($nextChar === "'" || $nextChar === '"') {
+                    if ($nextChar !== $quoteType) {
+                        $this->deprecations[] = sprintf("Character %s at position %d does not require to be escaped.", $nextChar, $i + 2);
+                    }
+                    $result .= $nextChar;
+                    $i++;
+                } elseif ($nextChar === 'x' && $i + 2 < $length && ctype_xdigit($str[$i + 1]) && ctype_xdigit($str[$i + 2])) {
+                    $result .= chr(hexdec($str[$i + 1] . $str[$i + 2]));
+                    $i += 2;
```

Comment:
Not sure if this is correct. Does Twig even support it? https://github.com/twigphp/Twig/issues/3604